### PR TITLE
grandpa: minor cleanup for gossip round start instant

### PR DESCRIPTION
**Pull request by andresilva**
Moved the gossip round start instant to the `LocalView` type and fixed the round duration constant.

### CODE REVIEW

File: client/finality-grandpa/src/communication/gossip.rs [hunk 1]
```diff
@@ -1121,8 +1136,10 @@ impl<Block: BlockT> Inner<Block> {
 	/// underlying gossip layer, which should happen every 30 seconds.
 	fn round_message_allowed<N>(&self, who: &PeerId, peer: &PeerInfo<N>) -> bool {
 		let round_duration = self.config.gossip_duration * ROUND_DURATION;
-		let round_elapsed = self.round_start.elapsed();
-
+		let round_elapsed = match self.local_view {
+			Some(ref local_view) => local_view.round_start.elapsed(),
+			None => return false,
+		};
```
Comment (octol) [inline client/finality-grandpa/src/communication/gossip.rs:1142]: The `None` case together with `self.config.is_authority`, would that give the same behaviour as the old code?

Comment (andresilva) [reply client/finality-grandpa/src/communication/gossip.rs:1142]: Overall there is the following change of behavior here: previously the round start was not behind any kind of optional value, therefore it had to be initialized on gossip creation with `Instant::now()` even though the round had not started yet. With the current approach we will only have round start defined after we have initialized the voter, therefore in the meantime all gossip messages will be disallowed. In practice this shouldn't change anything though, as the voter gets immediately initialized with a round after node startup.

File: client/finality-grandpa/src/communication/gossip.rs [hunk 2]
```diff
@@ -110,7 +110,7 @@ const CATCH_UP_THRESHOLD: u64 = 2;
 const PROPAGATION_ALL: u32 = 4; //in rounds;
 const PROPAGATION_ALL_AUTHORITIES: u32 = 2; //in rounds;
 const PROPAGATION_SOME_NON_AUTHORITIES: u32 = 3; //in rounds;
-const ROUND_DURATION: u32 = 4; // measured in gossip durations
+const ROUND_DURATION: u32 = 2; // measured in gossip durations
```
Comment (octol) [inline client/finality-grandpa/src/communication/gossip.rs:113]: Did the previous value trigger incorrect behaviour? I see it's used in two places below for checking if messages are allowed?

Comment (andresilva) [reply client/finality-grandpa/src/communication/gossip.rs:113]: No, it meant that we were filtering gossip messages for longer than we expected (e.g. we propagate to all peers after `ROUND_DURATION * PROPAGATION_ALL`), the actual behavior shouldn't change though. A GRANDPA round takes two timeout turns, i.e. prevote and precommit, and not 4.

**Pull request merged**